flake8_quotes/__init__.py

```diff
@@ -223,19 +233,38 @@ def get_quotes_errors(self, file_contents):
                 }
             # Otherwise (string is inline quote)
             else:
-                # If our string is a known good string, then ignore it
-                #   (')foo' -> good (continue)
-                #   "it(')s" -> good (continue)
-                #   (")foo" -> possibly bad
-                if self.config['good_single'] in unprefixed_string:
+                #   'This is a string'       -> Good
+                #   'This is a "string"'     -> Good
+                #   'This is a \"string\"'   -> Good
+                #   'This is a \'string\''   -> Bad (Q004)  Escaped inner quotes
+                #   '"This" is a \'string\'' -> Good        Changing outer quotes would not avoid escaping
+                #   "This is a string"       -> Bad (Q000)
+                #   "This is a 'string'"     -> Good        Avoids escaped inner quotes
+                #   "This is a \"string\""   -> Bad (Q000)
+                #   "\"This\" is a 'string'" -> Bad (Q000)
+                
+                middle = unprefixed_string[1:-1]
+                
+                # Correct outer quote characters
```

Comment:
This comment sound like we're actually correcting the string. Can we update it to a conditional instead?

```python
# If our string is our preferred type, then verify it doesn't only have escaped quotes of the same type
```